dbprocessing/dbprocessing.py

```diff
@@ -259,7 +258,16 @@ def figureProduct(self, filename=None):
         claimed = []
         for code, desc, arg, product in act_insp:
             try:
-                inspect = imp.load_source('inspect', code)
+                if sys.version_info < (3,4):
+                    import imp # imp is deprecated at python 3.4
+                    inspect = imp.load_source('inspect', code)
+                else:
+                    import importlib.util  # used in python 3.4 and above
+                    import importlib.machinery
+                    loader = importlib.machinery.SourceFileLoader('inspect', code)
+                    spec = importlib.util.spec_from_file_location('inspect', code, loader=loader)
+                    inspect = importlib.util.module_from_spec(spec)
+                    loader.exec_module(inspect)
```

Comment:
This is used in several places, see comments below.

I'm always wary of explicit self.version_info checks and would prefer we check, say, if importlib.util exists. However, the very next thing we're going to do is rip out support for old versions of Python, so I'm inclined to leave this as you wrote it.